src/lib/snyk-to-html.ts

```diff
@@ -177,6 +186,62 @@ async function readInputFromStdin(): Promise<string> {
   });
 }
 
+interface ActionableRemediation {
+  upgradeFrom: string; // e.g. package@version
+  upgradeTo: string;   // e.g. package@version
+  vulns: Vuln[];
+  severity: string;
+  severityScore: number;
+}
+
+interface Vuln {
+  id: string;
+  title: string;
+  severity: string;
+}
+
+function getVuln(id, vulnerabilities: any): Vuln {
+  const vuln = vulnerabilities.find((v) => v.id === id);
+  return {
+    id: vuln.id,
+    title: vuln.title,
+    severity: vuln.severity,
+  };
+}
+
+function getSeverityScore(vulns: Vuln[]) {
+  return (vulns.reduce((acc, vuln) => acc + (severityMap[vuln.severity] + 1), 0));
+}
+
+function getHighestSeverity(vulns: Vuln[]) {
+  const allVulns = vulns.map(vuln => severityMap[vuln.severity]);
+  allVulns.sort((a, b) => b - a);
+  const highestSev = Object.entries(severityMap).filter((i) => i[1] === allVulns[0])[0][0];
```

Comment:
My small brain is finding this hard to follow 😄 

Because there are only 3 severities, I'd be tempted to just loop over each vuln and return immediately if you find a vuln with a high severity, then check for medium, otherwise return low, like so:

```typescript
for (const vuln of vulns) {
  if (vuln.severity === 'high') {
    return 'high';
  }
  if (vuln.severity === 'medium') {
    return 'medium';
  }
}
return 'low';
```

This way you only iterate over vulns once if the first vuln you find is high

Reply:
That means it's over-complicated, and you make a great suggestion - will refactor!

Reply:
Just realised actually this function is redundant, as it finds the highest vuln severity to display on the label for each fix, but this element has been removed as it doesn't conform with the web app layout.